### ISSUE TITLE
How to use the ADDHISTORY hook?

### ISSUE THREAD

**Issue by SuperSandro2000** [akinomyoga/ble.sh#185]
I want to use the ``ADDHISTORY`` [hook](https://github.com/akinomyoga/ble.sh/wiki/Manual-%C2%A71-Introduction#122-function-blehook-hook_nameshell-command-v04-set-hooks) to filter my history and transform it but I couldn't find much documentation on it other than that it exists or even how to use it. Could you give a brief example how the hook is supposed to be used and if and how it can be used to transform history entries with bash substitution? Thanks!

**Comment by akinomyoga** [comment]
Ah, OK. There is a regression caused by [the recent change](https://github.com/akinomyoga/ble.sh/issues/179#issuecomment-1048416510). I have fixed it 479795d.

----

The usage is listed in the above code section where `hook_name` and `shell-command` may be replaced by `ADDHISTORY` and the function name. The function receives one argument to which the command line that will be added to the history will be specified.

```bash
# Example
blehook ADDHISTORY=myfunction
function myfunction { [[ $1 != ' '* ]]; }
```

Transformation is not supported. If you would like to add something different from the actually executed command, you can directly add the history entry by `history -s "transformed command"` and cancel the original history addition:

```bash
# Example
blehook ADDHISTORY=myfunction
function myfunction {
  history -s "$(transform $1)"
  return 1
}
```

**Comment by akinomyoga** [comment]
Can I close the issue? If you still have problems or questions, please reopen it.
